nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1041,6 +1043,78 @@ async fn project_policy_update(
 
 // IP Pools
 
+/// List all IP Pools that can be used by a given project.
+#[endpoint {
+    method = GET,
+    path = "/v1/ip-pools",
+    tags = ["projects"],
+}]
+async fn ip_pool_list_project_scoped(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    query_params: Query<PaginatedByNameOrId<params::ProjectSelector>>,
+) -> Result<HttpResponseOk<ResultsPage<IpPool>>, HttpError> {
+    // Per https://github.com/oxidecomputer/omicron/issues/2148
+    // This is currently the same list as /v1/system/ip-pools, that is to say,
+    // IP pools that are *available to* a given project, those being ones that
+    // are not the internal pools for Oxide service usage. This may change
+    // in the future as the scoping of pools is further developed, but for now,
+    // this is literally a near-duplicate of `ip_pool_list`:
+    let apictx = rqctx.context();
+    let handler = async {
+        let nexus = &apictx.nexus;
+        let query = query_params.into_inner();
+        let pag_params = data_page_params_for(&rqctx, &query)?;
+        let scan_params = ScanByNameOrId::from_query(&query)?;
+        let paginated_by = name_or_id_pagination(&pag_params, scan_params)?;
+        let opctx = crate::context::op_context_for_external_api(&rqctx).await?;
+        let pools = nexus
+            .ip_pools_list(&opctx, &paginated_by)
```

Comment:
I'd expect this to fail. The permissions of a user calling this endpoint inside of a silo and a user calling it at the system level should be different. If it doesn't, that's a bug.

I would recommend creating an app layer function called `project_ip_pool_list` in the `projects` app layer file. The resource that's being acted on is a synthetic resource called `IP_POOL_LIST` as seen below. 

https://github.com/oxidecomputer/omicron/blob/891acc7db7d798a914007df70fd1ae412de2141f/nexus/db-queries/src/db/datastore/ip_pool.rs#L56

We'll also need to update the policy for that synthetic resource to allow folks with project access to list the endpoint. You may want to go ahead and resolve the project to a lookup and pass that down to the app layer even if it's not used yet. (The code below is how to resolve the query to a lookup). 

https://github.com/oxidecomputer/omicron/blob/891acc7db7d798a914007df70fd1ae412de2141f/nexus/src/external_api/http_entrypoints.rs#L2869

I know there's a lot here, but I'm happy to help pair on it if you need.